app/src/main/java/org/onionshare/android/ui/OnionNotificationManager.kt

```diff
@@ -12,59 +13,95 @@ import android.os.Build.VERSION.SDK_INT
+        getForegroundNotification(app.getText(R.string.sharing_notification_title)).let {
+            nm.notify(NOTIFICATION_ID_FOREGROUND, it)
+        }
+        if (!isActive) getSharingNotification().let {
+            nm.notify(NOTIFICATION_ID_SHARING, it)
+        }
+    }
+
+    fun onStopped() {
+        nm.cancel(NOTIFICATION_ID_FOREGROUND)
+        nm.cancel(NOTIFICATION_ID_SHARING)
+    }
+
     fun getForegroundNotification(): Notification {
         val title = app.getText(R.string.starting_notification_title)
         val text = app.getText(R.string.starting_notification_text)
-        return getNotification(title, text)
-    }
-
-    fun onSharing() {
-        val n = getNotification(app.getText(R.string.sharing_notification_title))
-        nm.notify(NOTIFICATION_ID, n)
+        return getForegroundNotification(title, text)
     }
 
-    private fun getNotification(title: CharSequence, text: CharSequence? = null): Notification {
-        val pendingIntent: PendingIntent = Intent(app, MainActivity::class.java).let { i ->
-            val pendingFlags = if (SDK_INT < 23) 0 else FLAG_IMMUTABLE
-            PendingIntent.getActivity(app, 0, i, pendingFlags)
-        }
-        return NotificationCompat.Builder(app, CHANNEL_ID)
+    private fun getForegroundNotification(title: CharSequence, text: CharSequence? = null): Notification {
+        return NotificationCompat.Builder(app, CHANNEL_ID_FOREGROUND)
             .setContentTitle(title)
             .apply { if (!text.isNullOrBlank()) setContentText(text) }
             .setSmallIcon(R.drawable.ic_notification)
             .setColor(getColor(app, R.color.purple_onion_light))
             .setContentIntent(pendingIntent)
-            .setPriority(NotificationCompat.PRIORITY_LOW)
+            .setOngoing(true)
+            .setShowWhen(false)
+            .setPriority(PRIORITY_LOW)
             .setCategory(CATEGORY_SERVICE)
             .build()
     }
 
+    private fun getSharingNotification(): Notification {
+        return NotificationCompat.Builder(app, CHANNEL_ID_SHARING)
+            .setContentTitle(app.getText(R.string.notification_sharing_title))
+            .setContentText(app.getText(R.string.notification_sharing_text))
+            .setSmallIcon(R.drawable.ic_notification)
+            .setColor(getColor(app, R.color.purple_onion_light))
+            .setContentIntent(pendingIntent)
+            .setAutoCancel(true)
+            .setPriority(PRIORITY_HIGH)
+            .setCategory(CATEGORY_STATUS)
+            .build()
+    }
+
     @RequiresApi(26)
     private fun createNotificationChannels() {
-        val name = app.getString(R.string.sharing_channel_name)
-        val channel = NotificationChannel(CHANNEL_ID, name, IMPORTANCE_LOW).apply {
-            setShowBadge(false)
+        NotificationChannel(CHANNEL_ID_FOREGROUND, app.getString(R.string.notification_channel_name_foreground),
+            IMPORTANCE_LOW).apply { setShowBadge(false) }.let {
```

Comment:
I have a sneaking suspicion that you're trying to take me on a guided tour of Kotlin keywords...